### PR TITLE
add posix time formatting

### PR DESCRIPTION
**Pull request by kostiakoval**
Add posix functions for getting and formatting time. 
This function is required in RP #122.

### CODE REVIEW

File: Sources/POSIX/ctime.swift [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+ See http://swift.org/CONTRIBUTORS.txt for Swift project authors
+*/
+
+import func libc.time
+import func libc.ctime
+import typealias libc.time_t
+
+public func ctime() throws -> String {
+    var time = 0
+    libc.time(&time)
```
Comment (aciidb0mb3r) [inline Sources/POSIX/ctime.swift]: warn unused result 😛

File: Sources/POSIX/ctime.swift [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    let result = libc.ctime(&time)
+
+    return String(cString: result)
```
Comment (aciidb0mb3r) [inline Sources/POSIX/ctime.swift]: Shouldn't return type be `Double`

Comment (aciidb0mb3r) [inline Sources/POSIX/ctime.swift]: okay I see its not unix ts

File: Sources/POSIX/gmtime.swift [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+extension StringError: CustomStringConvertible {
+    public var description: String {
+        switch self {
+        case .NotEnoughSpace:
+            return "gmtime error: not enough space, increase resultSize."
+        }
+    }
+}
+
+public func gmtime(format: String, resultSize: Int = 200) throws -> String {
```
Comment (bhargavg) [inline Sources/POSIX/gmtime.swift]: I guess instead of returning `String`, it should return a `struct` of the time components.

Comment (kostiakoval) [inline Sources/POSIX/gmtime.swift]: I've split `gmtime` and `strftime`.  Now gmtime returns `tm` structure.
And `strftime` does standard posix string formatting :)

**Comment by kostiakoval** [comment]
After studying this topic I think it would be better to Use Thread Safe Functions
`gmtime_r` and etc. Any feedback on that idea?

http://linux.die.net/man/3/gmtime

**Comment by mxcl** [comment]
Probably, yes. Currently we are single thread, but that will change as Swift gains concurrency primitives.

**Comment by kostiakoval** [comment]
Done, use TS functions and added error handling.

**Comment by mxcl** [comment]
@swift-ci Please test

**Comment by kostiakoval** [comment]
test are passing, shall we merge it?

**Comment by mxcl** [comment]
Please 👍🏻

**Pull request merged**